### PR TITLE
Add ingest_resources for use with UI

### PR DESCRIPTION
**Pull request by sgillies**
Also remove a few params that look unnecessary.
[sc-57856]

### CODE REVIEW

File: src/tiledb/cloud/bioimg/ingestion.py [hunk 1]
```diff
@@ -381,7 +386,7 @@ def register_dataset_udf(
         *args,
         name=f"{dag_name} ingestor ",
         expand_node_output=input_list_node,
-        resources=DEFAULT_RESOURCES if resources is None else resources,
+        resources=resources or ingest_resources or DEFAULT_RESOURCES,
```
Comment (sgillies) [inline src/tiledb/cloud/bioimg/ingestion.py:389]: @ktsitsi Users of `ingest(resources={...})` without the `as_batch()` wrapper will get the behavior they expect. At the same time, our UI can specify `ingest_resources` and `as_batch()` will pass it through. How does that sound? I'll make the corresponding change upstream in the UI repo.

File: src/tiledb/cloud/bioimg/ingestion.py [hunk 2]
```diff
@@ -400,8 +405,6 @@ def register_dataset_udf(
             namespace=namespace,
             name=f"{dag_name} registrator ",
             expand_node_output=ingest_list_node,
-            resources=DEFAULT_RESOURCES if resources is None else resources,
```
Comment (sgillies) [inline src/tiledb/cloud/bioimg/ingestion.py:403]: Unless I'm missing something, registration doesn't need special resources. Does this look okay @ktsitsi ?

File: src/tiledb/cloud/bioimg/ingestion.py [hunk 3]
```diff
@@ -400,8 +405,6 @@ def register_dataset_udf(
...
-            image_name=DEFAULT_IMG_NAME,
```
Comment (sgillies) [inline src/tiledb/cloud/bioimg/ingestion.py:404]: Same for the image, I think. The basic image should work, yes?

File: src/tiledb/cloud/bioimg/ingestion.py (hunk 2 above)
Comment (ktsitsi) [reply src/tiledb/cloud/bioimg/ingestion.py:403]: For this one the architecture we are using is dynamic batch taskgraphs. I don't know if changes the resources between the different stages will lead to downscales of running pods or sth. Mainly I had it this way to be sure that the different stages correspond to the same node resources. This is a nice exercise though to test.

Comment (ktsitsi) [reply src/tiledb/cloud/bioimg/ingestion.py:403]: Indeed though you are right the registration stage is supposed to be lightweight so no special resources are generally needed.

File: src/tiledb/cloud/bioimg/ingestion.py [hunk 4]
```diff
@@ -219,7 +219,8 @@ def ingest_tiff_udf(
             "png": Converters.PNG,
         }.get(converter, Converters.OMETIFF)
 
-        experimental_reader = kwargs.get("experimental_reader", False)
+        experimental_reader = kwargs.pop("experimental_reader", False)
```
Comment (sgillies) [inline src/tiledb/cloud/bioimg/ingestion.py:222]: @ktsitsi I found that if I didn't pop off this keyword argument, `from_bioimg()` could receive it twice.

**Comment by sgillies** [comment]
This works. The extra resources are only needed for the task that touches the imagery. Prep and registration tasks work fine with minimal resources.

Untitled

**Pull request merged**